pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -632,25 +631,36 @@ func (m *kubeGenericRuntimeManager) updateContainerResources(pod *v1.Pod, contai
 		pod, containerSpec = restoredPod, restoredContainer
 	}
 
-	var containerResources runtimeapi.LinuxContainerResources
+	memoryLimit := int64(0)
+	cpuShares := int64(0)
 	cpuRequest := int64(0)
 	cpuLimit := int64(0)
-	memoryLimit := int64(0)
+	var containerResources runtimeapi.LinuxContainerResources
+
+	if memoryLimits, found := resources.Limits[v1.ResourceMemory]; found {
+		memoryLimit = memoryLimits.Value()
+	}
+	containerResources.MemoryLimitInBytes = memoryLimit
```

Comment:
shouldn't we add a default value to memoryLimit in case the if statement on 640 didn't find it in the map

Reply:
if the zero default value is okay, please add a comment to specify it.

Reply:
Default value is 0. I tested it for Guaranteed and Burstable ( cpu only, memory only) cases for increase and decrease. There is the open issue about decreasing CPU which has to do with ordering. I'll work on a fix for that hopefully sometime soon.